src/OpenLoco/src/Drawing/SoftwareDrawingContext.cpp

```diff
@@ -1378,10 +1378,82 @@ namespace OpenLoco::Drawing
         // @return width @<cx>
         static uint16_t getStringWidthNewLined(const char* buffer)
         {
-            registers regs;
-            regs.esi = X86Pointer(buffer);
-            call(0x00495715, regs);
-            return regs.cx;
+            auto font = *_currentFontSpriteBase;
+            uint16_t maxWidth = 0;
+
+            for (auto* ptr = buffer; *ptr != '\0'; ++ptr)
+            {
+                uint16_t lineWidth = 0;
+                for (; *ptr != '\0'; ++ptr)
```

Comment:
Should we split up the inner for loop to another function, so we have a function that computes a just single line width? Then this 'outer' function can take the max of the width of all lines.

Reply:
Yeah that might be neater.